app/models/configuration_source.rb

```diff
@@ -0,0 +1,6 @@
+class ConfigurationSource < ApplicationRecord
+  # An Tower project/repo has many playbooks (configuration_scripts)
+  has_many    :configuration_scripts, dependent => :destroy
+  belongs_to  :manager
+  belongs_to  :scm_reference
```

Comment:
`scm_reference` is a `git_reference`.  A project/repo would need a `git_reference` or `svn_reference`, right?  (The `git_reference` can accommodate a `svn` if we can ignore the name. Or we can create base classes `scm_reference` and `scm`)